Update scraper for new aimpoint mon page

jeanconn opened the pull request:
## Description

Update scraper for new aimpoint mon page


## Interface impacts
<!-- API changes, file format updates, coordination of changes with the community. -->

## Testing
<!-- If relevant describe any special setup for testing. -->

### Unit tests
<!-- At least one of these must be checked if unit tests exist. DELETE the unchecked/untested options. -->
- [x] No unit tests


### Functional tests
<!-- Describe and document results of any functional tests, otherwise leave the text below -->

```
ska3-jeanconn-fido> git rev-parse HEAD
d6ff9b290f18376ed57b3d01013cf8f41d809921
ska3-jeanconn-fido> python twiki_wg/ssawg_trending_scraper.py --data-dir /proj/sot/ska/www/ASPECT_ICXC/test_review_outputs/twiki-wg/twiki-wg-pr32
```
With output to

https://icxc.cfa.harvard.edu/aspect/test_review_outputs/twiki-wg/twiki-wg-pr32/ssawg_trending.html

taldcroft commented on twiki_wg/ssawg_trending_scraper.py:
```diff
@@ -114,8 +114,9 @@ def parse_page(self):
             for local_link in self.soup.find_all("a"):
-                temp = local_link["href"]
-                local_link["href"] = self.url + temp
+                if "href" in local_link:
+                    temp = local_link["href"]
+                    local_link["href"] = self.url + temp
```
This is pointing to a mistake in `aimpoint_mon`. In https://github.com/sot/aimpoint_mon/commit/f3e5e2c07b947d3122de3384d9444fa962c7b524 the anchor tag should be done with:
```
<h3 id="aimpoint-trending"> Observed aimpoint differences trend </h3>
```

taldcroft replied:
Using `name` in an `<a>` tag is deprecated and doesn't exist in HTML5. (I didn't know that of course but that's what google says).

jeanconn replied:
I figured we probably had plenty of other "a" without hrefs as possible candidates for this local link code but OK.  And I'd figure if we move to using anchors with id that needs new logic here doesn't it?

taldcroft replied:
> I figured we probably had plenty of other "a" without hrefs as possible candidates for this local link code
A search of the `sot` org for `<a name` shows a number of matches but only aimpoint_mon is a candidate for scraping. I wasn't previously aware of this issue but we should just move to using the `id` tag in place of `name` going forward.

> if we move to using anchors (as id in header element) that needs new logic here doesn't it?
The scraping code does not care about anchors, right? I guess I don't follow you here.

jeanconn replied:
"The scraping code does not care about anchors, right?"  Maybe my language is wrong -- I thought this change was in a section that is just looping over <a> / anchor tags  -- and I don't completely see where the code would build up a link to the new "<h3 id="aimpoint-trending">" location you suggest but heck, it looks like it was just *broken* with what I tried here anyway I and I didn't notice in the test output.

taldcroft reviewed (commented):
Probably should fix aimpoint_mon and revert that code change.

jeanconn commented:
I fixed aimpoint_mon and reverted the code change here that was to work around  `<a name="aimpoint-trending"></a>`.

Trying to just get the "paragraph" copied into the trending page that describes the intraobservation aimpoint drift with this code I think shows that this ssawg scraper code is not really maintainable code?  I updated to just using next next next to get the pieces and have the link from that Intra-observation aimpoint drift paragraph at least go somewhere (now it goes back to the h3 id on the original page instead of the local ssawg index page).

javierggt commented:
I am assuming the output of this is what we saw yesterday in SSAWG. Is that right?

jeanconn commented:
No, the aimpoint mon pages in the official ssawg trending page are still showing the old aimpoint_mon page not the new aimpoint_mon3 content (the most obvious different to the user is actually that the new page doesn't use the interactive plots at all).

jeanconn commented:
We did get an "ok" to move to using the new page at the 02-28-2024 meeting and the intent of this twiki PR was to finish the move to the new page.

taldcroft reviewed (approved):
Approved pending confirmation that the functional test output was generated with the latest version of this PR.

jeanconn commented:
Reran functional test output and update Description.  Output still looks good.
